In [5]:
# Import libraries
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

# Load the dataset
# Replace 'weather_data.csv' with the path to your dataset
df = pd.read_csv('weather_data.csv')

# Select features (X) and target variable (y)
X = df[['Temperature_C', 'Wind_Speed_kmh', 'Humidity_pct']]
y = df['Precipitation_mm']

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Train the linear regression model
model = LinearRegression()
model.fit(X_train, y_train)

# Make predictions on the testing set
y_pred = model.predict(X_test)

# Evaluate the model
mse = mean_squared_error(y_test, y_pred)
print('Linear Regression Mean Squared Error:', mse)


Linear Regression Mean Squared Error: 8.664671382787539


The Mean Squared Error (MSE) of 8.6647 obtained for the linear regression model indicates that, on average, the squared difference between the actual precipitation values and the predicted values is approximately 8.6647 square millimeters.

#### Mean Squared Error (MSE):

The MSE measures the average squared difference between the predicted and actual values. A lower MSE indicates better performance, as it means the model's predictions are closer to the actual values on average.
In this case, an MSE of 8.6647 suggests that the linear regression model provides relatively accurate predictions of precipitation based on the input features (temperature, wind speed, and humidity).
Interpretation:

The MSE value of 8.6647 is relatively low, indicating that the linear regression model is effective in capturing the underlying relationships between the input features and the target variable (precipitation).
The model's predictions are reasonably close to the actual precipitation values, suggesting that the chosen features (temperature, wind speed, and humidity) are informative for predicting precipitation.

In [6]:
from sklearn.ensemble import GradientBoostingRegressor

# Create a Gradient Boosting regressor
gb_model = GradientBoostingRegressor(random_state=42)

# Train the model
gb_model.fit(X_train, y_train)

# Make predictions on the testing set
y_pred_gb = gb_model.predict(X_test)

# Evaluate the model
mse_gb = mean_squared_error(y_test, y_pred_gb)
print('Gradient Boosting Mean Squared Error:', mse_gb)


Gradient Boosting Mean Squared Error: 8.63729433700543


The Mean Squared Error (MSE) of 8.6373 obtained for the Gradient Boosting model is slightly lower than the MSE obtained for the linear regression model. Here's an evaluation of the MSE and the model's performance:

#### Mean Squared Error (MSE):

The MSE measures the average squared difference between the predicted and actual values. A lower MSE indicates better performance, as it means the model's predictions are closer to the actual values on average.
In this case, an MSE of 8.6373 suggests that the Gradient Boosting model provides relatively accurate predictions of precipitation based on the input features (temperature, wind speed, and humidity).

#### Interpretation:

The lower MSE value compared to linear regression suggests that the Gradient Boosting model is better at capturing the underlying relationships between the input features and the target variable (precipitation).

The model's predictions are slightly closer to the actual precipitation values compared to the linear regression model, indicating improved performance.

